lib/adf_info.py

```diff
@@ -270,61 +296,84 @@ def __init__(self, config_file, debug=False):
+                    input_ts_locs = self.get_cam_info("cam_ts_loc", required=True)
+                    input_ts_loc = Path(input_ts_locs[case_idx])
+
+                    #Get years from pre-made timeseries file(s)
+                    syear, eyear = self.get_climo_yrs_from_ts(input_ts_loc, case_name)
+
+                    msg = f"Using first found year: {syear}\n"
+                    msg += f"and last found year: {eyear}"
+                    print(msg)
+                #End if
+            #End if
 
             #Check if history file path exists:
-            if cam_hist_locs:
+            if cam_hist_locs[case_idx]:
                 #Get climo years for verification or assignment if missing
                 starting_location = Path(cam_hist_locs[case_idx])
                 file_list = sorted(starting_location.glob('*'+hist_str+'.*.nc'))
                 #Partition string to find exactly where h-number is
                 #This cuts the string before and after the `{hist_str}.` sub-string
-                # so there will always be three parts: 
+                # so there will always be three parts:
                 # before sub-string, sub-string, and after sub-string
                 #Since the last part always includes the time range, grab that with last index (2)
                 #NOTE: this is based off the current CAM file name structure in the form:
                 #  $CASE.cam.h#.YYYY<other date info>.nc
                 case_climo_yrs = [int(str(i).partition(f"{hist_str}.")[2][0:4]) for i in file_list]
                 case_climo_yrs = sorted(np.unique(case_climo_yrs))
 
+                case_found_syr = int(case_climo_yrs[0])
+                case_found_eyr = int(case_climo_yrs[-1])
 
                 #Check if start or end year is missing.  If so then just assume it is the
                 #start or end of the entire available model data.
                 if syear is None:
-                    print(f"No given start year for {case_name}, using first found year...")
-                    syear = int(case_climo_yrs[0])
+                    msg = f"No given start year for {case_name}, "
+                    msg += f"using first found year: {case_found_syr}"
+                    print(msg)
+                    syear = case_found_syr
                 elif (syear) not in case_climo_yrs:
                     msg = f"Given start year '{syear}' is not in current dataset "
                     msg += f"{case_name}, using first found year: {case_climo_yrs[0]}\n"
                     print(msg)
-                    syear = int(case_climo_yrs[0])
+                    syear = case_found_syr
+                else:
+                    syear = int(syear)
                 #End if
                 if eyear is None:
-                    print(f"No given end year for {case_name}, using last found year...")
-                    eyear = int(case_climo_yrs[-1])
+                    msg = f"No given start year for {case_name}, "
```

Comment:
I think this should be `end year`:

```suggestion
                    msg = f"No given end year for {case_name}, "
```